server/etcdserver/server.go

```diff
@@ -2082,6 +2090,9 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 			}
 			lg.Panic("failed to create snapshot", zap.Error(err))
 		}
+
+		verifyConsistentIndexIsLatest(lg, snap, s.consistIndex.ConsistentIndex())
```

Comment:
Since there isn't any real key/value data in v2 store anymore, probably it doesn't make much sense to perform the snapshot in a separate goroutine any more; then the cindex should be equal to snapshot index, accordingly we should use `verifySnapshotIndex` directly instead of adding a new function `verifyConsistentIndexIsLatest`